File: frontend/cypress/test-solutions/top-k-frequent-elements.js
class Solution {
    topKFrequent(nums, k) {
        // Count frequency of each number
        const count = new Map();
        for (const num of nums) {
            count.set(num, (count.get(num) || 0) + 1);
        }
        
        // Sort by frequency and take top k
        return Array.from(count.keys())
            .sort((a, b) => count.get(b) - count.get(a))
            .slice(0, k);
    }
}

File: backend/test-solutions/top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        // Count frequency of each number
        unordered_map<int, int> count;
        for (int num : nums) {
            count[num]++;
        }
        
        // Use min heap to keep track of k most frequent elements
        priority_queue<pair<int, int>, vector<pair<int, int>>, greater<pair<int, int>>> heap;
        
        for (auto& p : count) {
            heap.push({p.second, p.first});
            if (heap.size() > k) {
                heap.pop();
            }
        }
        
        // Extract result
        vector<int> result;
        while (!heap.empty()) {
            result.push_back(heap.top().second);
            heap.pop();
        }
        
        return result;
    }
};